lib/widgets/rectangular_inset_shadow_box.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/widgets.dart';
+
+/// A widget that overlays rectangular inset shadows on a child.
+///
+/// The use case of this is casting shadow on scrollable UI elements.
+/// For example, when there is a list of items, the shadow could be
+/// a visual indicator for over scrolled areas.
+///
+/// See also:
+///  * https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3860-11890&node-type=frame&t=oOVTdwGZgtvKv9i8-0
+class RectangularInsetShadowBox extends StatelessWidget {
```

Comment:
"rectangular box" seems redundant — a box is necessarily rectangular. I think InsetShadowBox would be a fine name.

Or I guess this is a box with rectangular inset shadows (the shadows are rectangular). But I think an "inset shadow" on a box, which is a rectangle, is also going to be rectangular.

It's true this doesn't cover as wide a range of cases as CSS `box-shadow: inset`, as discussed at https://github.com/zulip/zulip-flutter/pull/990#discussion_r1792373624 . I think that reflects the CSS feature growing to be wider than its name fits. To clarify that this doesn't cover that CSS feature, the clearest thing would be to just mention it in the doc.